src/webclients/components/sdfv/vscode_sdfv.ts

```diff
@@ -518,7 +535,15 @@ export class VSCodeSDFV extends SDFV {
         return this.sdfgString;
     }
 
-    public getMetaDict(): { [key: string]: any } | null {
+    public getMetaDict(): { [key: string]: any } {
+        if (!this.sdfgMetaDict) {
+            // If SDFG property metadata isn't available, use the static one and
+            // query an up-to-date one from DaCe if available.
+            vscode.postMessage({
+                type: 'dace.query_sdfg_metadata',
+            });
+            return staticSdfgMetaDict;
+        }
```

Comment:
you should also try to query the one from our website perhaps? I don't know what's more up to date, but this is fine as is too. Please add your opinion here.

Reply:
Yes, that is indeed a good idea. My proposal is the following order (pseudocode):

```
try:
	dace.get_metadata()
catch dace offline:
	try:
		github_pages.get_metadata()
	catch no connection:
		use locally cached metadata
```

Reply:
Yes, sounds good. Will you add it in this PR? This is also good because metadata is tied to the dace release rather than the vscode plugin release